evaporate.js

```diff
@@ -885,7 +886,7 @@
                 }
                 var completedAt = new Date(u.completedAt || FAR_FUTURE);
 
-                return con.partSize === u.partSize && completedAt > HOURS_AGO;
+                return con.partSize === u.partSize && completedAt > HOURS_AGO && me.name === u.awsKey && con.bucket === u.bucket;
```

Comment:
If you've uploaded the same file to a different location previously, even when `allowS3ExistenceOptimization` is true, it still generates requests to the AWS key of the first time you uploaded the file, not the second - then instantly completes the upload (but to the old key)